### PR TITLE
Fix Edit Command 

### PR DESCRIPTION
**Pull request by sophiien**
- Checking of person subtypes is now done using instanceof 
- User cannot edit person / module if not on respective module viewing page

### CODE REVIEW

**Review by benmurphyy — CHANGES_REQUESTED** [review]
## Bugs Found
- when i type in certain incomplete commands, like `edit 1`, and press enter, nothing happens, no message is shown to me. I can only see in the terminal that there was a nosuchelementexception thrown. Please make sure all possible erroneous inputs will display some message to the user informing them of the problem.
Screenshot 2022-03-15 at 11 34 54 AM

- i can type any `edit 1 o/<gibberish>` on the person page, instead of telling me that thats invalid, it tells me to switch view to module. Probably because of an if-else block used in your parsing that has too large a scope.
- adding a new module to a person dosen't work at all, like `edit 1 o/person nm/cs2103`, nothing happens when i press enter. instead this error is showing in terminal.
Screenshot 2022-03-15 at 11 39 22 AM

### note regarding your update to this pr - please include the commands and how to use them in pr description for easy debugging, or update the user guide to contain them. Make sure that each work before updating this pr

**Review by geraldneo567 — CHANGES_REQUESTED** [review]
You need to throw either a `CommandException` or a `ParseException` depending on whether you are in `EditCommand::execute()` or `EditCommandParser`, so that when the user enters wrong input (e.g input that is missing required information or has the wrong formatting, i.e when `!(argMultimap.getValue(PREFIX_XXXX).isPresent()))` you should throw an exception instead of just continuing execution. This will cause the user to be prompted and see what's wrong on the GUI.

**Comment by sophiien** [comment]
****Usage of edit command (updated in UG too) 

- edit INDEX o/OPTION [n/NAME or MODULENAME] [p/PHONE] ...
- e.g. edit 1 o/person nm/CS2103 : adds CS2103 to list of person's modules 

The following incorrect commands will result in an error message shown in the console:
- Missing compulsory fields i.e. INDEX or OPTION 
- Invalid module code 
- Editing person/module when not in correct viewing page 
- Less than 1 field provided to be edited

File: docs/UserGuide.md [hunk 1]
```diff
@@ -139,21 +139,20 @@ Examples:
...
 * Edits the entity type defined by `o/OPTION`. This is a compulsory field.
-* Depending on the entity type, certain fields will be compulsory.
-* Before choosing to edit module or person, user can see which `INDEX` to edit by changing the UI to show the relevant list
-  * `list o/module` : To display list of modules with respective indexes
-  *  `list o/person` : To display list of persons with respective indexes 
+* Before choosing to edit module or person, user can see which `INDEX` to edit by changing the UI to show the relevant list. User will not be allowed to edit if not on the correct page. 
+  * `list v/modules` : To display list of modules with respective indexes
+  * `list v/people` : To display list of persons with respective indexes 
 * 2 values for `OPTION`:
     * `module`: Edits the module specified by compulsory field `m/MODULE`. Optional fields `[n/NAME] [nm/NEWMODULE]` to specify the new name or module code of the module.
```
Comment (benmurphyy) [inline docs/UserGuide.md:149]: I think you wrote the wrong fields here, this should be for person, not module

**Review by benmurphyy — CHANGES_REQUESTED** [review]
Seems to be working, good job!
Just clean up user guide and the test case thats failing, and we can merge!

File: docs/UserGuide.md (hunk 1 above)
Comment (benmurphyy) [reply docs/UserGuide.md:149]: this part of the user guide is still not fixed in this pr

**Review by benmurphyy — CHANGES_REQUESTED** [review]

File: src/main/java/unibook/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -191,10 +232,11 @@ public CommandResult execute(Model model, Boolean isPersonListShowing,
 
             // Find all profs and students with this module and will be edited
             for (Person person : lastPersonList) {
-                Set<Module> moduleSet = person.getModules();
+                Set<Module> moduleSet = person.getModulesModifiable();
                 if (moduleSet.contains(moduleToEdit)) {
                     moduleSet.remove(moduleToEdit);
                     moduleSet.add(editedModule);
+                    person.setModules(moduleSet);
```
Comment (yokester98) [inline src/main/java/unibook/logic/commands/EditCommand.java:239]: I feel that this line
`person.setModules(moduleSet)`
is redundant actually, as the person's moduleSet is already beinng modified in the lines above. If my understanding is correct, the person's moduleSet will still be updated without needing this line. What do you think? 🤔

**Review by yokester98 — CHANGES_REQUESTED** [review]
Well done! I did notice a portion of the code that may be redundant though (but I might be wrong!) Take a look and let me know what you think?

**Review by geraldneo567 — APPROVED** [review]
LGTM!

**Review by yokester98 — APPROVED** [review]

**Review by benmurphyy — APPROVED** [review]

**Pull request merged**